Launcher.py

```diff
@@ -254,10 +278,17 @@ def _stop(self, *largs):
 
     Launcher().run()
 
+    # avoiding Launcher reference leak
+    # and don't try to do something with widgets after window closed
+    global refresh_components
+    refresh_components = None
+
 
 def run_component(component: Component, *args):
     if component.func:
         component.func(*args)
+        if refresh_components:
+            refresh_components()
```

Comment:
Since `run_component` is only used for drag&drop and startup, I don't think this is a problem, but it seems odd to `refresh_component()` for all `func`s, not just the `install_apworld` one. I don't have a great suggestion though. Could raise a custom exception in component.func that signals a reload, could give component.func and optional return value, or could give an extra attribute to component.

Reply:
I think an extra refresh once in a while will not be bad.
I wasn't able to tell that a refresh was happening when I dragged the patches in.
I can imagine there might be other component functions in the future that might also need the refresh.

Reply:
Yeah, we can potentially "fix" it in the future if we need to.